coffeescript-loader/test.js

```diff
@@ -1,28 +1,33 @@
-import { ok } from 'assert';
-import { spawn } from 'child_process';
+import { fileURLToPath } from 'url';
+import { ok, strictEqual } from 'assert';
+import { spawnSync } from 'child_process';
 import { execPath } from 'process';
 
 // Run this test yourself with debugging mode via:
 // node --inspect-brk --experimental-loader ./loader.js ./fixtures/esm-and-commonjs-imports.coffee
 
-const child = spawn(execPath, [
-  '--experimental-loader',
+const loader = fileURLToPath(new URL(
   './loader.js',
+  import.meta.url
+));
+const fixture = fileURLToPath(new URL(
   './fixtures/esm-and-commonjs-imports.coffee',
-]);
-let stdout = '';
-child.stdout.setEncoding('utf8');
-child.stdout.on('data', (data) => {
-  stdout += data;
-});
-let stderr = '';
-child.stderr.setEncoding('utf8');
-child.stderr.on('data', (data) => {
-  stderr += data;
-});
+  import.meta.url
+));
 
-child.on('close', (code, signal) => {
-  ok(stdout.includes('Hello from CoffeeScript'), 'Main entry transpiles');
-  ok(stdout.includes('HELLO FROM ESM'), 'ESM import transpiles');
-  ok(stdout.includes('Hello from CommonJS!'), 'Named CommonJS import fails');
-});
+const { status, stderr, stdout } = spawnSync(
+  execPath,
+  [
+    '--experimental-loader',
+    loader,
+    fixture,
+  ],
+  { encoding: 'utf8' },
+);
+
+console.error(stderr);
+
+strictEqual(status, 0);
+ok(stdout.includes('Hello from CoffeeScript'), 'Main entry transpiles');
+ok(stdout.includes('HELLO FROM ESM'), 'ESM import transpiles');
+ok(stdout.includes('Hello from CommonJS!'), 'Named CommonJS import fails');
```

Comment:
I know this was in the original, but “fails”?

Reply:
Ummm, idk. maybe copy+pasta?